### PR TITLE
feat: Add CI

### PR DESCRIPTION
**Pull request by AntonBazhal**

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+  workflow_dispatch:
+
+jobs:
+  Setup-Env:
+     runs-on: ubuntu-latest
```
Comment (JasperAtD2L) [inline .github/workflows/main.yml:11]:
```suggestion
     runs-on: [self-hosted, AWS, Linux]
```

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+     container: bash
```
Comment (JasperAtD2L) [inline .github/workflows/main.yml:12]: This shouldn't need a container

File: .github/workflows/main.yml [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+     timeout-minutes: 1
+     outputs:
+       isMain: ${{ steps.setupEnv.outputs.isMain }}
+     steps:
+       - name: Setup Env
+         id: setupEnv
+         uses: Brightspace/actions-parse-workflow-env@v2
+
+  Tests:
+    runs-on: ubuntu-latest
```
Comment (JasperAtD2L) [inline .github/workflows/main.yml:22]:
```suggestion
    runs-on: [self-hosted, AWS, Linux]
```

File: .github/workflows/main.yml [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+
+      - name: Lint
+        run: npm run lint
+
+      - name: License Test
+        run: npm run license
+
+  Release:
+    if: ${{ needs.Setup-Env.outputs.isMain == 'true' }}
+    runs-on: ubuntu-latest
```
Comment (JasperAtD2L) [inline .github/workflows/main.yml:50]:
```suggestion
    runs-on: [self-hosted, AWS, Linux]
```

File: .github/workflows/main.yml [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+        with:
+          persist-credentials: false
+
+      - name: Setup Node
+        uses: Brightspace/third-party-actions@actions/setup-node
+        with:
+          node-version: 12
+
+      - name: Semantic Release
+        uses: BrightspaceUI/actions/semantic-release@master
```
Comment (JasperAtD2L) [inline .github/workflows/main.yml:65]: Guessing the default for this is npmjs.or and not artifactory.

File: .github/workflows/main.yml (hunk 1 above)
Comment (AntonBazhal) [reply .github/workflows/main.yml:11]: this is a public repo, no need to use self-hosted runners

File: .github/workflows/main.yml (hunk 5 above)
Comment (AntonBazhal) [reply .github/workflows/main.yml:65]: this is a public repo, so we should use NPM

Comment (JasperAtD2L) [reply .github/workflows/main.yml:64]: Yup, just making sure our homebrew action defaults to npmjs.org

File: .github/workflows/main.yml (hunk 1 above)
Comment (JasperAtD2L) [reply .github/workflows/main.yml:11]: Right 👍

**Review by JasperAtD2L — APPROVED** [review]

**Pull request merged**